projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/__tests__/manual-oauth2-flow-component.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import {render, fireEvent, waitFor, cleanup} from '@testing-library/react';
+import { requestAPI } from '../handler';
+import {ManualOAuth2Flow} from "../components/Login2";
+import React from "react";
+
+jest.mock('../handler', () => ({
+    requestAPI: jest.fn()
+}));
+
+
+describe('ManualOAuth2Flow Component', () => {
+
+    beforeEach(() => {
+        window.open = jest.fn();
+    })
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        cleanup()
+    });
+
+    it('should render without crashing', () => {
+        render(<ManualOAuth2Flow/>);
+    });
+
+    it('should open a new authentication window on button click', async () => {
+        (requestAPI as jest.MockedFunction<typeof requestAPI>).mockResolvedValue('http://auth-url.com');
+        const component = render(<ManualOAuth2Flow/>);
+        const button = component.getByText("Click here to start.");
+        fireEvent.click(button);
```

Comment:
I am using fireEvent for consistency. But from what I am reading, overall [recommendations](https://ph-fritsche.github.io/blog/post/why-userevent) are to use user-event instead - https://testing-library.com/docs/user-event/intro/#differences-from-fireevent